File: src/main/java/me/miki/shindo/api/websocket/AccountType.java
package me.miki.shindo.api.websocket;

import java.util.Locale;

public enum AccountType {
    LOCAL,
    MICROSOFT,
    OFFLINE;

    public static AccountType from(String raw) {
        if (raw == null) {
            return LOCAL;
        }
        String normalized = raw.trim().toUpperCase(Locale.ROOT);
        for (AccountType type : values()) {
            if (type.name().equals(normalized)) {
                return type;
            }
        }
        return LOCAL;
    }

    public String getWireValue() {
        return name();
    }
}

File: src/main/kotlin/me/miki/shindo/api/websocket/AccountType.kt
package me.miki.shindo.api.websocket

import java.util.*

enum class AccountType {
    LOCAL,
    MICROSOFT,
    OFFLINE;

    fun getWireValue(): String = name

    companion object {
        @JvmStatic
        fun from(raw: String?): AccountType {
            if (raw == null) return LOCAL
            val normalized = raw.trim().uppercase(Locale.ROOT)
            for (type in entries) {
                if (type.name == normalized) return type
            }
            return LOCAL
        }
    }
}
